### DISCUSSION TITLE
New External Guest User Created- Not Getting ALL Items (Count) from Default Search Result {LocalSharepointResults}

### DISCUSSION THREAD

**Discussion by RamBalani** [microsoft-search/pnp-modern-search#2700] [Q&A] [answered]
We have one PRODUCTION site intended to be   accessed by Existing and New External Guest Users.
 from the same site {LocalSharepoitnResults} as result source for PnP Web Part in both cases, i.e existing vs new External Guest.

When accessing PnP Search Webparts-- on the initial load of Pnp Search Results  web part -{LocalSharepointResults} shows the scope of items.

HOWEVER:

1-While Existing External Guest Users  access to the  PnP Search results  Web Parts show 27688 items from LocalSharepointResults

2The New External Guest we just created which also  accesses the same  PnP Search WebPart on the same site  only shows a number far less than 27688

So to be clear --- the issue is that # 1 and # 2 above BOTH  gets a DIFFERENT total items count while accessing the same site and using the same GuestSearch.Aspx which contains the  SAME PnP Search Web parts.

What could explain this ? 

Is this Permission Based trimming we are neglecting and  not configuring correctly or mere indexing  time to wait for indexing on a new External Guest account ?

The items count from LocalSharepointResults when the New External Guest is logged in seems to rise in time soon after the External Guest account was created, it this  subject to local search indexing maybe over time? 

 The Result Source in PnP Web Part is set to "LocalSharepointResult" and our Query Template is {SearchTerms} Path:{SiteCollection}

Can anyone offer some feedback as to what is going on? or What we can do to resolve if a mis-step on our parts if that be the case?

Ram Balani

**Comment by kasperbolarsen** [comment]
HI @RamBalani , first of all lets find out if the potential error is in SharePoint search or in PnP.There are a number of ways to do this:
1) Install the SP Query Tool and run the query from there by the two external users
2) Install the SP Editor chrome/edge extention and run the query from there
3) Have the users go to [tenant]/search and use the query Path:[the URL of the original Site Collection]
4) Go to the Search Result Sources on the Site Collection, add a new one with the Path:{SiteCollection} and use that Search result source in a CLASSIC search web part ( on a web part page) 

If the records count is identical then the issue is with the PnP solution, otherwise the issue is with either permissions or SPO search.

Let us know your findings.

**Reply by RamBalani** [reply]
Thanks Kaspebolarsen for your feedback here. THat said-- you maybe missing a key point which is my fault for NOT making crystal clear when I originally posted this issue.

The query template or query process is , as I see it , irrelevant, i.e. since the LocalSharepointResults configured in PnP Search  as I understand it, is pegged to Site Collection and SIte level Result Source pages, both are designated as 'Default'  with 'LocalSharepointresults' on both levels, one and the same..
 
Therefore any  actual  query intended by the user  has nothing to do  with total items count difference that is shown before ANY query is in fact entered, i.e. to be clear the 'LocalSharepointResults' count which varies for Existing vs New External Guest appears BEFORE any queries whatsoever upon PnP Search Web part page load.

No query tool, if I am correct, will reveal anymore information as to your point of whether PnP or Sharepoint Search is at fault here.

It is (to me at least) likely that  this is a  Sharepoint result source artifact  issue or something else specially since -- and this is the puzzling piece to me - - the 'LocalSharePointResults' count FOR THE NEW EXTERNAL GUEST ACCOUNT JUST CREATED  seems to be rising in TIME  (around 1600 now  from yesterday's earlier 203  without us changing anything) vs OLD existing External Guest total items count of 27688 on the 'LocalSharepointResults' count. (27688 BTW matches what an ORG Admin/Member also gets with the same site/same PnP access)

Clearly-no query has been issued nor any user filters selected at the point the anomaly /count issue shows up.

The count(s) shown appears as soon as the page containing the PnP Search page appears by virtue of the result source configured on the PnP Search web part as 'LocalSharepointresults' . The total item count shows up  immediately after a user  clicks the link to our GuestSearch.aspx which presents the page where the PnP Search Parts are deployed that contain the LocalSharepointResults configured as result source out of the box so to speak. 

This is a  mystery that has nothing to do with either any query(cause NONE has been entered!) nor PnP Search part (since the one and same PnP Web part is used by both Existing and New External Guest ) YET the LocalSharepointResults count vary.  Follow ?

THanks for your attention and further input if any.

**Reply by kasperbolarsen** [reply]
Hi @RamBalani , right. So the issue has nothing to do with PnP Modern Search but is a general SharePoint Search question? 

I guess you have already triggered a request for a reindexing of content on the site collation using the button at /_layouts/15/srchvis.aspx ?  ( it sure sounds like an indexing issue)

In the case that a reindexing didn't solve the issue I would suggest either raising a support ticket with MS on the tenant in question, or ask the question in a SharePoint Search forum, like https://learn.microsoft.com/en-us/answers/questions/

**Comment by RamBalani** [comment, marked as answer]
See below-thanksf for the follow-up - -much appreciated.

https://tinyurl.com/2p9ee4ba
[cid:ca7536aa-c41c-4fbe-b66d-a72cc7142cca]

________________________________

**Reply by RamBalani** [reply]
Hi @RamBalani , right. So the issue has nothing to do with PnP Modern Search but is a general SharePoint Search question? **[Ram]** yes -that is correct.

I guess you have already triggered a request for a reindexing of content on the site collation using the button at /_layouts/15/srchvis.aspx ? ( it sure sounds like an indexing issue)  **[Ram]** yes-we have fired of an Indexing request -one of the first things we did in fact-no positive results however to a resolution.

In the case that a reindexing didn't solve the issue I would suggest either raising a support ticket with MS on the tenant in question, or ask the question in a SharePoint Search forum, like https://learn.microsoft.com/en-us/answers/questions/   **[Ram]** We have done that-raise a Microsoft Support Ticket-and I di raise a question n the Microsoft community forum as well.
